src/components/faceVerification/standalone/hooks/useGiveUpDialog.js

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useContext } from 'react'
+import { isUndefined, negate, pickBy } from 'lodash'
+
+import { fireEvent, FV_GIVEUP } from '../../../../lib/analytics/analytics'
+import AsyncStorage from '../../../../lib/utils/asyncStorage'
+import { useDialog } from '../../../../lib/dialog/useDialog'
+import GiveUpDialog from '../components/GiveUpDialog'
+import { FVFlowContext } from '../context/FVFlowContext'
+import { requestIdle } from '../../../../lib/utils/system'
+import useFVRedirect from './useFVRedirect'
+
+const useGiveUpDialog = (navigation, type) => {
+  const { showDialog } = useDialog()
+  const fvRedirect = useFVRedirect()
+  const { isFVFlow } = useContext(FVFlowContext)
+  const { navigate } = navigation
+
+  const onReasonChosen = useCallback(
+    (reason = undefined) => {
+      const data = pickBy({ reason }, negate(isUndefined))
```

Comment:
what is this complicated expression for? simply to check that reason is defined?

Reply:
ehh yes. and its just a left over of the original implementation
didn't look over it for any refactoring..